src/Maestro/Microsoft.DotNet.Maestro.Tasks.Tests/GetBuildManifestMetadataTests.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Linq;
+using System.Threading;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace Microsoft.DotNet.Maestro.Tasks.Tests
+{
+    [TestFixture]
+    public class GetBuildManifestMetadataTests
+    {
+        private PushMetadataToBuildAssetRegistry pushMetadata;
+        private string testManifestFolderPath = @".\TestManifests\"; 
+
+        public GetBuildManifestMetadataTests()
+        {
+            pushMetadata = new PushMetadataToBuildAssetRegistry();
+        }
+        [Test]
+        public void EmptyManifestFolderPath()
+        {
+            Action act = () => pushMetadata.GetBuildManifestsMetadata("", new CancellationToken());
+            act.Should().Throw<ArgumentException>().WithMessage("The path is empty. (Parameter 'path')");
+        }
+
+        [Test]
+        public void ParseBasicManifest()
+        {
+            var data = pushMetadata.GetBuildManifestsMetadata(testManifestFolderPath + "OneManifest", new CancellationToken());
```

Comment:
What is "data"?  The "var" and then the "Item1" makes it hard to understand what is actually happening here.

Reply:
I completely agree, I personally wouldn't have picked a three item Tuple for the return type. I'll call it out explicitly though, since that's what it is.

Reply:
I updated it to use the newer Tuple handling and gave everything in the Tuple a name, so it should be clearer now.